### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by joseivanlopez**
Requires this for tests: https://github.com/yast/yast-yast2/pull/761

### CODE REVIEW

File: src/lib/services-manager/clients/services_manager.rb [hunk 1]
```diff
@@ -37,6 +37,11 @@
 
 module Y2ServicesManager
   module Clients
+    # Services Manager client
+    #
+    # TODO: The whole ServicesManager client should be rewritten to use CWM dialogs, widgets, etc,
+    # and the logic should be splitted into the proper classes instead of having an only one client
```
Comment (imobachgs) [inline src/lib/services-manager/clients/services_manager.rb]: I would remove *an*.

File: src/lib/services-manager/service_loader.rb [hunk 2]
```diff
@@ -178,6 +178,9 @@ def extract_services
       service_names.zip(ss).each do |name, s|
         sh = services[name] # service hash
         if s
+          sh[:service] = s
```
Comment (imobachgs) [inline src/lib/services-manager/service_loader.rb]: I will have to solve some conflicts in a near future :)

File: test/clients/services_manager_test.rb [hunk 3]
```diff
@@ -22,20 +22,76 @@
...
-  # Finds in the widgets tree a widget with the given id
+  # Checks whether a widgets tree contains the given id
   #
   # @param tree [Yast::Term]
   # @param id [Symbol]
   #
   # @return [Boolean]
-  def exist_widget?(tree, id)
-    !tree.nested_find { |w| w.is_a?(Yast::Term) && w.value == :id && w[0] == id }.nil?
+  def contain_widget?(tree, id)
```
Comment (imobachgs) [inline test/clients/services_manager_test.rb]: These methods look like candidates to be extracted to yast2 helpers. It is out of the scope of this PR, but we should take it into account.

File: test/clients/services_manager_test.rb [hunk 4]
```diff
@@ -81,45 +162,61 @@ def exist_widget?(tree, id)
...
 
-    let(:units_output) do
-      [
-        "sshd.service  loaded active   running OpenSSH Daemon\n",
-        "postfix.service loaded inactive dead    Postfix Mail Agent\n"
-      ]
+    let(:selected_service_name) { "sshd" }
+
+    context "when the selected service is running" do
+      it "shows an 'stop' button" do
```
Comment (imobachgs) [inline test/clients/services_manager_test.rb]: an -> a

File: test/clients/services_manager_test.rb [hunk 5]
```diff
@@ -81,45 +162,61 @@ def exist_widget?(tree, id)
...
+    context "when the selected service is not running" do
+      let(:selected_service_name) { "postfix" }
 
-    let(:sshd_service) do
-      double("service",
-        name:     "sshd",
-        enabled?: true,
-        active?:  true
-      )
+      it "shows an 'start' button" do
```
Comment (imobachgs) [inline test/clients/services_manager_test.rb]: an -> a

File: test/services_manager_service_test.rb [hunk 6]
```diff
@@ -221,5 +263,46 @@ def stub_services
...
+      end
+    end
+
+    describe "#substate" do
+      it "returns the service substate" do
+        expect(service.substate("sshd")).to eq("running")
+      end
+
+      context "if the service is not found" do
+        let(:service_name) { "unkown" }
```
Comment (imobachgs) [inline test/services_manager_service_test.rb]: typo: unkown -> unknown

File: test/clients/services_manager_test.rb (hunk 3 above)
Comment (joseivanlopez) [reply test/clients/services_manager_test.rb]: Yes, they would be helpful.

**Pull request merged**